### ISSUE TITLE
Team permissions are not updated for restricted users

### ISSUE THREAD

**Issue by Masgalor** [go-gitea/gitea#13055]
- Gitea version (or commit ref): 1.12.5
- Git version: 2.20.1
- Operating system: Debian stable
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
Assuming the following setup:

Organization_A owns Repo_A.
Organization_A also has two Teams "Owners" and "ReadOnly".
The team "ReadOnly" is set as collobarotor and has read-only access to Repo_A.

If a new restricted user is created and added to Organization_A's team "ReadOnly" the new user can see all repositories owned by Organization_A but can't access any of them (gets a 404). Only after removing the team and adding it back as collobarotor, the new user is able to see the content of Repo_A.
If the organization has several repositories every single one must be removed and reassigned to the team to update the permissions for the new user.

I tried to reproduce this on try.gitea.io but I can't create a restricted user there.
An unrestricted user could see the content of the repository immediately after joining the team.

Settings for team "ReadOnly"
![team settings](https://user-images.githubusercontent.com/30129213/95223440-e2e76a00-07f9-11eb-9218-ed0596a37d3f.PNG)

**Comment by evolunis-ws** [comment]
I had an unrelated case, where a supposed malfunction was caused by the repositories being public.
I had to make repositories "private" for the team-specific access restrictions to become effective.

**Comment by gabschne** [comment]
Removing all repositories from the team, adding a single one and then all others with the "Add all" button did the trick for me. (It may work without adding a single one first, I'm not sure because I tried a lot of things.)

**Comment by 6543** [comment]
should be fixed as of now

please open a new issue if an similar issue do ocure
